packages/headless/src/controllers/quickview/headless-quickview.ts

```diff
@@ -0,0 +1,76 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {fetchResultContent} from '../../features/result-preview/result-preview-actions';
+import {
+  ConfigurationSection,
+  ResultPreviewSection,
+} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+
+export interface QuickviewProps {
+  /**
+   * The options for the `Quickview` controller.
+   */
+  options: QuickviewOptions;
+}
+
+export interface QuickviewOptions {
+  /**
+   *
+   */
+  result: Result;
+}
+
+export interface Quickview extends Controller {
+  /**
+   * Retrieves the preview content for the configured result.
+   */
+  fetchResultContent(): void;
+
+  /**
+   * The state for the `Quickview` controller.
+   */
+  state: QuickviewState;
+}
+
+export interface QuickviewState {
+  /**
+   * The result preview HTML content.
+   *
+   * @default ""
+   */
+  content: string;
+}
+
+/**
+ * Creates a `Quickview` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `Quickview` properties.
+ * @returns A `Quickview` controller instance.
+ */
+export function buildQuickview(
+  engine: Engine<ConfigurationSection & ResultPreviewSection>,
+  props: QuickviewProps
+): Quickview {
+  const {dispatch} = engine;
+  const controller = buildController(engine);
+  const uniqueId = props.options.result.uniqueId;
+
+  return {
+    ...controller,
+
+    fetchResultContent() {
+      dispatch(fetchResultContent({uniqueId}));
+    },
+
+    get state() {
+      const preview = engine.state.resultPreview;
+      const content = uniqueId === preview.uniqueId ? preview.content : '';
+
+      return {
+        content,
+      };
+    },
+  };
+}
```

Comment:
Is there a way to know if a result has a quickview (i.e. a Html preview)?

For example, passing a youtube video result would not yield one? Is it possible to know this before making a request?

Reply:
Perhaps whether the "index body of an item is empty" or not https://coveo.github.io/search-ui/components/quickview.html#options.alwaysshow

Reply:
There's flags on the result.

```
"flags" : "HasHtmlVersion",
"hasHtmlVersion" : true,
```

That's what should be used. We should not be replicating the "always show" thingy from the JSUI, that was a big mess.

Reply:
Exactly what I was looking for, thanks. I also see a `hasMobileHtmlVersion`. I guess this just means that the html is responsive? I don't see how one would request a mobile vs. desktop preview?